core/src/main/java/com/klaviyo/coresdk/KlaviyoConfig.kt

```diff
@@ -0,0 +1,34 @@
+package com.klaviyo.coresdk
+
+data class KlaviyoConfig (
+        val apiKey: String,
+        val networkTimeout: Int,
+        val networkFlushInterval: Int
+    ) {
+    class Builder {
+        private val NETWORK_TIMEOUT_DEFAULT: Int = 500
+        private val NETWORK_FLUSH_INTERVAL_DEFAULT: Int = 60000
+
+        private lateinit var apiKey: String
```

Comment:
By making this a `lateinit` var without a default value like the others I'm forcing the builder to have this variable. Any `KlaviyoConfig` that is built out without a supplied `apiKey` will through an `UninitializedProperyAccessException`. Maybe not the perfect way to do this but I thought it was a start to forcing some key pieces of the configuration that we can't provide defaults for

Reply:
hmm, i like the idea and its a super clever way of doing validation but i think we should provide a more specific exception

Reply:
another option could be to require the apiKey in the Builder constructor

Reply:
maybe all non-defaultable fields can live in the constructor? thoughts?

Reply:
Had the same thought on adding it to the Builder constructor but I wondered what other non-defaultables we might have in the future, and then destroying the purpose of even having the Builder in the first place if many of the params are just being passed into the constructor. I'm probably overthinking it though/

Reply:
yeah thats true, agreed not a good pattern to follow if there are many non-defaults, found this article thats interesting https://dev.to/siy/when-builder-is-anti-pattern-3j92